Lib/pathlib/types.py

```diff
@@ -290,7 +300,12 @@ def glob(self, pattern, *, recurse_symlinks=True):
         select = globber.selector(parts)
         return select(self.joinpath(''))
 
-    def walk(self, top_down=True, on_error=None, follow_symlinks=False):
+    def walk(
+        self,
+        top_down: bool = True,
+        on_error: Optional[Callable[[Exception], None]] = None,
+        follow_symlinks: bool = False,
+    ) -> Generator[tuple[Self, list[str], list[str]]]:
```

Comment:
Isn't it possible to simply use Iterable here as well?

Reply:
I switched to Iterator